api/external/infra_proxy/request/environments.proto

```diff
@@ -5,11 +5,22 @@ option go_package = "github.com/chef/automate/api/external/infra_proxy/request";
 
 import "google/protobuf/struct.proto";
 
+message SearchQuery {
+	// The search query used to identify a list of items.
```

Comment:
This needs a lot more information on what constitutes a query.
Your PR preamble gives some...
"... has to be in the format *:*, name:* or name:chef-env-* etc."
 but that is insufficient. What do you mean by "etc." ? Must there always be a colon? Can there be more than one colon. If a wildcard can exist in `*:*` then surely we could also do "*:chef-env", but there is no example of that.
It is OK to have examples, but also need to explain the requirements.
What I see from this small set of examples:
1. Format must be `x:y` where x is ?? and y is ??.
2. `x` may be a field name (what fields??) or a wildcard.
3. `y` may be a complete value (what values??), a value prefix ending with a wildcard, or a wildcard by itself.
That is just my inference; please adjust this to accurately reflect the requirements.
_(This applies to several PRs and several proto files in each PR)._

Reply:
So here in the environment list, currently targeting for searching with the name only.

Have taken reference from chef-manage so if nothing is provided for search use `*:*` for wildcard

otherwise use `name:SEARCH_PATTERN` like example.




For more context on search pattern  chef search support `key:search_pattern` ref: https://docs.chef.io/chef_search/#query-syntax

where `key` is a field name that is found in the JSON description of an indexable object on the Chef Infra Server (a role, node, client, environment, or data bag) and search_pattern defines what will be searched for, using one of the following search patterns: exact, wildcard, range, or fuzzy matching. Both key and search_pattern are case-sensitive; `key` has limited support for multiple character wildcard matching using an asterisk ("*") (and as long as it is not the first character).